Make README more newbie friendly

osg opened the pull request:
Not sure how to test to see if this link behaves properly...

Commit:
Refer out to tip about human readible output

Commit:
Add hyphen

Commit:
Break apart each step to make it easier on newbies

Commit:
Change "In The Wild" for non-native English reader

osg commented:
Note that I omitted one book for the purpose of eliminating one extra scroll.

osg commented on README.md:
```diff
@@ -3,66 +3,79 @@ kindle-highlights
 * [tobi/highlights](https://github.com/tobi/highlights) - Download your Kindle highlights and email random ones to your inbox
 
-### Contributing to kindle-highlights (PR's welcome)
+### Contributing to kindle-highlights (PRs welcome)
```
Refer to http://www.grammarbook.com/punctuation/apostro.asp.

speric replied:
I fixed this typo on `master`; thanks!

osg replied:
Not sure how to resolve the merge conflict that derives from this change...

osg commented:
Happy to make tweaks based on your feedback, @speric.

Already received feedback from a friend that this is not intended to be run within `irb`. However, it is indeed useful to see if it is working, so perhaps we can include both. Again, focusing on step-by-step for a newbie.

denisdefreyne commented on README.md:
```diff
@@ -3,66 +3,79 @@ kindle-highlights
-# pass in your Amazon credentials. Loads your books (not highlights)
-# on init, so might take a while
-kindle = KindleHighlights::Client.new("email.address@gmail.com", "password")
+1. In an interactive Ruby shell ([`irb`](http://ruby-doc.org/stdlib-2.0.0/libdoc/irb/rdoc/IRB.html)), run the first of several commands:
```
Since this is a library, it’s not intended to be run with irb. (It _can_ be, but mostly for playing around.) I’d rephrase this as:

``````
1. Load the library:

    ```
    require 'kindle_highlights'
    ```
``````

denisdefreyne commented on README.md:
```diff
@@ -3,66 +3,79 @@ kindle-highlights
-    "startLocation" => 29496,
-    "timestamp"     => 1320901233000
-  },
+1. Provide your Amazon credentials to access your book titles:
```
“Construct a `KindleHighlights::Client` with your Amazon credentials:”

opensourcegrrrl replied:
Less newbie friendly, but ok.

osg replied:
Why does the user need to know that it creates a `KindleHighlights::Client`?

denisdefreyne replied:
The user needs to explicitly create a `KindleHighlights::Client`, so the documentation should say so.

denisdefreyne commented on README.md:
```diff
@@ -3,66 +3,79 @@ kindle-highlights
+
+   `kindle = KindleHighlights::Client.new("email.address@gmail.com", "password")`
+   
+   This might take a while.
```
Needs details on why this takes a while. “This fetches information for all Kindle books, and might therefore take a while.”

denisdefreyne commented on README.md:
```diff
@@ -3,66 +3,79 @@ kindle-highlights
+   
+   This might take a while.
+   
+1. Get each book (where the key is the [ASIN](http://www.amazon.com/gp/seller/asin-upc-isbn-info.html)), and its corresponding title:
```
Suggestion: “Call `#books` to get a hash of all books, where the key is the ASIN and the value the book title.”

Otherwise, it looks like a key/ASIN is somehow needed to get the list of books.

denisdefreyne commented on README.md:
````diff
@@ -3,66 +3,79 @@ kindle-highlights
-]
-```
+    ```
+1. (Optional) If you are not sure how to produce human-readable output as in the preceding example, refer to [Best way to pretty print a hash]( http://stackoverflow.com/questions/8842546/best-way-to-pretty-print-a-hash).
````
I’d remove this.

denisdefreyne commented on README.md:
````diff
@@ -3,66 +3,79 @@ kindle-highlights
-```
+    ```
+1. (Optional) If you are not sure how to produce human-readable output as in the preceding example, refer to [Best way to pretty print a hash]( http://stackoverflow.com/questions/8842546/best-way-to-pretty-print-a-hash).
+1. Get your highlights for a specific book by using its ASIN:
````
“… by calling `#highlights_for`, passing in the book’s ASIN:”

denisdefreyne commented on README.md:
````diff
@@ -3,66 +3,79 @@ kindle-highlights
+  ```
 
-### In The Wild
+### Alternatives
````
This list collects projects using `kindle-highlights`; it’s not a list of alternatives.

denisdefreyne commented on README.md:
````diff
@@ -3,66 +3,79 @@ kindle-highlights
+   `kindle.books`
+   
+   Example output is as follows:
+    ```
````
Add the programming language after the three backticks, so that it gets highlighted properly:

``````
    ```ruby
    {
        "B002JCSCO8" => "The Art of the Commonplace: The Agrarian Essays of Wendell Berry",
``````

denisdefreyne commented on README.md:
````diff
@@ -3,66 +3,79 @@ kindle-highlights
+  
+  Example output is as follows:
+
+  ```
````
Add the programming language here too.

osg commented:
Thanks, @ddfreyne! Will come back to this tomorrow after some sleep.

Commit:
Update README.md

Commit:
Add rationale as to why it might take a while

Commit:
Intentionally hide non-WYSIWYG logic gap

Commit:
Add programming language to code snippets

Commit:
Use plain language to define the section;

Keep in plural for style reason; if a new project is added, the section does not need to change. It’s a small documentation maintenance best practice.

osg commented:
Removed [WIP]. This is ready for final review.

osg commented:
Please walk me through resolving this merge conflict.

denisdefreyne commented:
I can help you out later.

denisdefreyne commented on README.md:
```diff
@@ -3,66 +3,79 @@ kindle-highlights
+
+1. Load the library:
+
+   `require 'kindle_highlights'`
```
Prefer triple backticks for marking up code blocks:

``````
    ```
    require 'kindle_higlights'
    ```
``````

(same elsewhere)

denisdefreyne commented on README.md:
```diff
@@ -3,66 +3,79 @@ kindle-highlights
+
+   `require 'kindle_highlights'`
+
+1. Provide your Amazon credentials to access your book titles:
```
Provide your Amazon credentials how?

“to access your book titles” is not the point of creating a `KindleHighlights::Client`, so I’d leave that out.

osg commented:
Hi @speric, should I stop working on this PR and close it? Looks like the README is getting updated in parallel in `master` and I am not sure if the changes in my PR are for nought.

Commit:
Use code block markdown consistently

speric commented:
@opensourcegrrrl I made a few changes in the gem itself that necessitated updating the README. This work had begun before you opened your PR. I'm sorry that there's now a merge conflict with your branch, but I couldn't release a new version of the gem without documenting the changes, as they were breaking changes.

Anyway, it's up to you if you want to rebase and work on the new README or not. I don't want to give you work to do, but neither do I want to discourage you from doing something you want to do!

I also would like to say that I appreciated your changes and the direction you were going. I did not realize until recently that many people use this gem who don't write Ruby, and I assumed too much familiarity with Ruby in the README.

I am going to make further changes to the README in the next day or two that are directly influenced by this PR. Specifically, I liked the way you moved explanations of what each method does, and what it returns, out of a comment within the Markdown code block. It reads better than the way I had it. Also, I now capture one of the mysterious errors you mentioned in another issue, and print out a readable explanation of what happened. In a future change to the README I will document how to work around that error.

So, thank you!

osg commented:
@speric Ok, great. If you can wrap up your changes by Sat night, say, I can hopefully incorporate all of the feedback from Denis by Sunday night Berlin time. If not, that is okay too.

Kinda want to wrap things up as "good enough" before heading on vacation to China for about ten days. I anticipate not being on my ’puter as much, and my internet connections are likely to be spotty.

Also, I can easily just create a new branch after your changes, and thereby forego dealing with merge conflicts. :)

osg commented:
Update: I have a healing fractured hand, and will need to revisit this in May. If you don't mind, please just keep this open so I can come back to it.

speric commented:
@opensourcegrrrl I am going to close this for now; feel free to re-open it in the future.

osg commented:
Yeah, no problem. Will need to come back to this due to priorities. Thanks for the note!
